cpp/oneapi/dal/backend/memory.hpp

```diff
@@ -80,6 +82,161 @@ class usm_deleter {
     sycl::queue queue_;
 };
 
+template <typename T>
+using unique_usm_ptr = std::unique_ptr<T, usm_deleter<T>>;
```

Comment:
```suggestion
using unique_usm_ptr = detail::unique<T, usm_deleter<T>>;
```
?

Reply:
Now I'm thinking about removing `detail::unique`, it seems useless...

Reply:
What about `detail::shared`?

Reply:
The same